kolibri/plugins/style_guide/assets/src/views/shell/table-of-content.vue

```diff
@@ -0,0 +1,64 @@
+<template>
+
+  <ul>
+    <li v-for="anchor in anchors">
+      <a :href="route" @click="goToAnchor(anchor.hash)">
+        {{ anchor.label }}
+      </a>
+    </li>
+  </ul>
+
+</template>
+
+
+<script>
+
+  /**
+   * A component for auto-generating a table of content for a page. All elements
+   * with an [id] attribute defined will be shown in the TOC.
+   */
+
+  const map = require('lodash/map');
+
+  module.exports = {
+    name: 'TableOfContent',
+    props: {
+      /** The route of the current page. E.g. "#button" */
+      route: {
+        type: String,
+      },
```

Comment:
Instead of passing it in as a prop, the current route [can be extracted from component](https://router.vuejs.org/en/api/route-object.html) using `this.$route`